src/soca/State/State.h

```diff
@@ -57,6 +57,10 @@ namespace soca {
       State(const State &);
       virtual ~State();
       State & operator=(const State &);
+      void transpose(const State & DistState, const eckit::mpi::Comm & global,
+         const int & ensNum, const int & transNum) {
+         throw eckit::NotImplemented("Soca State::transpose not implemented", Here());
+      }
```

Comment:
Some compilers (intel?) give warnings if there are unused named parameters, this is fixable by commenting out the names.

```suggestion
      void transpose(const State & /*DistState*/, const eckit::mpi::Comm & /*global*/, 
         const int & /*ensNum*/, const int & /*transNum*/) {
         throw eckit::NotImplemented("Soca State::transpose not implemented", Here());
      }
```